### PR TITLE
Better get_table_rows upper/lower bounds support

### LINKED ISSUE #3948: RPC get a table rows, is work bad!
Details:
**Summary:** Using eos RPC to get blockchain table data, the return data is not follow the condition!

**Description:**  I get the voters table data again by again, but it's add just a little row in the table, and it not return all data any time!

## Steps To Reproduce:
  1. I use the script to get table rows, it not be get all!
```
more = True
lens = -1
    while more:
        data = {"scope":"eosio", "code":"eosio", "table":"voters", "json": True}
        resp = requests.post("http://eosbattles.com/v1/chain/get_table_rows", json=data)
        data = resp.json()
```
  2. Then I add the `upper_bound` condition like this, and it not work:
```
 data = {"scope":"eosio", "code":"eosio", "table":"voters", "upper_bound": -1, "json": True}
```
  3. Then i also add a `lower_bound` condition like this, and it not work:
```
 data = {"scope":"eosio", "code":"eosio", "table":"voters", "upper_bound": -1, "lower_bound": 10000000, "json": True}
```
4. Then i also add a `limit` condition like this, and it not work:
```
 data = {"scope":"eosio", "code":"eosio", "table":"voters", "upper_bound": -1, "lower_bound": 10000000, "json": True, "limit": 10000000}
```

## Supporting Material/References:

  * https://eosio.github.io/eos/group__eosiorpc.html#v1chaingettablerows

## Impact

this is my all scripts:
```
import requests


def test():
    more = True
    lens = -1
    while more:
        data = {"scope":"eosio", "code":"eosio", "table":"voters", "json": True}
        resp = requests.post("http://eosbattles.com/v1/chain/get_table_rows", json=data)
        data = resp.json()
        rows = data.get("rows")
        print(len(rows))

if __name__ == '__main__':
    test()

```

### PR DESCRIPTION
**Pull request by heifner**
- Add support for name and symbol values passed to lower/upper bound

It is now possible to use account names in lower/upper bound for tables with account_name as primary key:
` ./cleos get table eosio eosio voters -L kevinh -U kevini`

Resolves #3948

### CODE REVIEW

**Comment by heifner** [comment]
Now only attempts conversion to name/symbol for uint64_t. Thanks @arhag

File: plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp [hunk 1]
```diff
@@ -50,6 +48,16 @@ struct permission {
 template<typename>
 struct resolver_factory;
 
+template<typename Type>
+Type convert_to_type(const string& str, const string& desc) {
+   try {
+      return fc::variant(str).as<Type>();
+   } FC_RETHROW_EXCEPTIONS(warn, "Could not convert {desc} string '${str}' to key type.", ("desc", desc)("str",str) )
```
Comment (arhag) [inline plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp]: `{desc}` should be `${desc}`. Other than that, it looks good.

Comment (heifner) [reply plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp]: ugh, that is what I get for hurrying.

**Comment by qiushaoxi** [comment]
cleos get table eosio eosio voters -L 111111222333 
if name is all num , it can not work

**Comment by heifner** [comment]
`./cleos get table eosio eosio voters -L " 111111222333"`

**Pull request merged**